v2/googlecloud-to-googlecloud/src/main/java/com/google/cloud/teleport/v2/templates/bigtablechangestreamstopubsub/schemautils/PubSubUtils.java

```diff
@@ -0,0 +1,396 @@
+        PubSubFields.SOURCE_INSTANCE,
+        (pb, chg) -> chg.getString(PubSubFields.SOURCE_INSTANCE.name()));
+    FORMATTERS.put(
+        PubSubFields.SOURCE_CLUSTER,
+        (pb, chg) -> chg.getString(PubSubFields.SOURCE_CLUSTER.name()));
+    FORMATTERS.put(
+        PubSubFields.SOURCE_TABLE, (pb, chg) -> chg.getString(PubSubFields.SOURCE_TABLE.name()));
+    FORMATTERS.put(
+        PubSubFields.TIEBREAKER, (pb, chg) -> chg.getInt(PubSubFields.TIEBREAKER.name()));
+
+    // Just in case, validate that every column in the enum has a formatter
+    for (PubSubFields field : PubSubFields.values()) {
+      Validate.notNull(FORMATTERS.get(field));
+    }
+  }
+
+  private final BigtableSource source;
+  private final PubSubDestination destination;
+  private transient Charset charsetObj;
+
+  public PubSubUtils(BigtableSource sourceInfo, PubSubDestination destinationInfo) {
+    this.source = sourceInfo;
+    this.destination = destinationInfo;
+    this.charsetObj = Charset.forName(sourceInfo.getCharset());
+  }
+
+  public BigtableSource getSource() {
+    return source;
+  }
+
+  public PubSubDestination getDestination() {
+    return destination;
+  }
+
+  private void readObject(ObjectInputStream in) throws IOException, ClassNotFoundException {
+    in.defaultReadObject();
+    charsetObj = Charset.forName(source.getCharset());
+  }
+
+  private String convertBase64ToString(String base64String) {
+    return new String(Base64.getDecoder().decode(base64String), charsetObj);
+  }
+
+  private byte[] convertBase64ToBytes(String base64String) {
+    return Base64.getDecoder().decode(base64String);
+  }
+
+  public PubsubMessage mapChangeJsonStringToPubSubMessageAsAvro(String changeJsonSting)
+      throws Exception {
+    String messageEncoding = this.getDestination().getMessageEncoding();
+    ChangelogEntryMessage changelogEntryMessage = new ChangelogEntryMessage();
+    JSONObject changeJsonParsed = new JSONObject(changeJsonSting);
+
+    changelogEntryMessage.setRowKey(
+        ByteBuffer.wrap(
+            (byte[]) FORMATTERS.get(PubSubFields.ROW_KEY_BYTES).format(this, changeJsonParsed)));
+    changelogEntryMessage.setModType(
+        ModType.valueOf(
+            (String) FORMATTERS.get(PubSubFields.MOD_TYPE).format(this, changeJsonParsed)));
+    changelogEntryMessage.setIsGC(
+        (Boolean) FORMATTERS.get(PubSubFields.IS_GC).format(this, changeJsonParsed));
+    changelogEntryMessage.setTieBreaker(
+        (Integer) FORMATTERS.get(PubSubFields.TIEBREAKER).format(this, changeJsonParsed));
+    changelogEntryMessage.setColumnFamily(
+        (String) FORMATTERS.get(PubSubFields.COLUMN_FAMILY).format(this, changeJsonParsed));
+    changelogEntryMessage.setCommitTimestamp(
+        (Long) FORMATTERS.get(PubSubFields.COMMIT_TIMESTAMP).format(this, changeJsonParsed));
+    changelogEntryMessage.setColumn(
+        ByteBuffer.wrap(
+            (byte[]) FORMATTERS.get(PubSubFields.COLUMN).format(this, changeJsonParsed)));
```

Comment:
Maybe we should rename PubSubFields.COLUMN to PubSubFields.COLUMN_BYTES for uniformity